ouroboros-network/src/Ouroboros/Network/Mux.hs

```diff
@@ -83,34 +98,32 @@ muxStart udesc wfn rfn sdufn close  rescb_m = do
                         | ptcl <- [minBound..maxBound]
                         , mode <- [ModeInitiator, ModeResponder] ]
 
-    mpsJob cnt pmss mpd = do
+    mpsJob cnt pmss udesc mpdId = do
+        let mpd = udesc mpdId
         w_i <- atomically newEmptyTMVar
         w_r <- atomically newEmptyTMVar
 
-        return [ mpdInitiator mpd (muxChannel pmss (mpdId mpd) ModeInitiator w_i cnt)
+        return [ mpdInitiator mpd (muxChannel pmss (AppProtocolId mpdId) ModeInitiator w_i cnt)
                      >> mpsJobExit cnt
-               , mpdResponder mpd (muxChannel pmss (mpdId mpd) ModeResponder w_r cnt)
+               , mpdResponder mpd (muxChannel pmss (AppProtocolId mpdId) ModeResponder w_r cnt)
                      >> mpsJobExit cnt
```

Comment:
@dcoutts had a question on this, weather `mpsJobExit` needs to be done using `finally`; the way it runs here it will only be executed in clean shutdown.

Reply:
It does not, when we fail we don't try to send all queued messages.

Reply:
Please just add a comment.

Reply:
I'll update the comment for `mpsJobExit`.